kubetest/e2e.go

```diff
@@ -626,7 +630,47 @@ func kubemarkTest(testArgs []string, dump, numNodes string) error {
 	// Run tests on the kubemark cluster.
 	if err := control.XmlWrap(&suite, "Kubemark Test", func() error {
 		testArgs = util.SetFieldDefault(testArgs, "--ginkgo.focus", "starting\\s30\\pods")
-		return control.FinishRunning(exec.Command("./test/kubemark/run-e2e-tests.sh", testArgs...))
+
+		// detect master IP
+		if err := os.Setenv("MASTER_NAME", os.Getenv("INSTANCE_PREFIX")+"-kubemark-master"); err != nil {
+			return err
+		}
+
+		masterIP, err := control.Output(exec.Command(
+			"gcloud", "compute", "addresses", "describe",
+			os.Getenv("MASTER_NAME")+"-ip",
+			"--project="+o.gcpProject,
+			"--region="+o.gcpZone[:len(o.gcpZone)-2],
+			"--format=value(address)"))
+		if err != nil {
+			return fmt.Errorf("failed to get masterIP: %v", err)
+		} else {
+			if err := os.Setenv("KUBE_MASTER_IP", strings.TrimSpace(string(masterIP))); err != nil {
+				return err
+			}
+		}
+
+		if os.Getenv("ENABLE_KUBEMARK_CLUSTER_AUTOSCALER") == "true" {
+			testArgs = append(testArgs, "--kubemark-external-kubeconfig="+os.Getenv("DEFAULT_KUBECONFIG"))
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		//util := os.Getenv("WORKSPACE") + "/kubernetes/cluster/kubemark/util.sh"
+		//testArgs = append([]string{"-c", "source", util, " ; ./hack/ginkgo-e2e.sh"}, testArgs...)
+		cmd := exec.Command("./hack/ginkgo-e2e.sh", testArgs...)
+		cmd.Env = append(
+			os.Environ(),
+			"KUBERNETES_PROVIDER=kubemark",
+			"KUBE_CONFIG_FILE=config-default.sh",
+			fmt.Sprintf("KUBECONFIG=%s/test/kubemark/resources/kubeconfig.kubemark", cwd),
```

Comment:
Seems like we're assuming here that cwd is k/k repo root.
If you're making that assumption anyway, why to use cwd? Can't you just use "." instead?

Reply:
because later on the relative path can change and it will complain about cannot find KUBECONFIG

Reply:
That's possible - let's keep it this way.